filters/block/block_test.go

```diff
@@ -234,9 +246,87 @@ func TestBlock(t *testing.T) {
 		}
 	})
 
+	t.Run("pass request with filter chain and check content", func(t *testing.T) {
+		content := "hello world"
+
+		be := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			res, err := stdlibio.ReadAll(r.Body)
+			r.Body.Close()
+			if err != nil {
+				w.WriteHeader(500)
+				w.Write([]byte("Failed to read body"))
+				return
+			}
+			if s := string(res); s != content {
+				t.Logf("backend received: %q", s)
+				w.WriteHeader(400)
+				w.Write([]byte("wrong body"))
+				return
+			}
+			w.WriteHeader(200)
+			w.Write([]byte("OK"))
+		}))
+		defer be.Close()
+
+		args := []interface{}{"foo"}
+		args2 := []interface{}{"bar"}
+		r := &eskip.Route{Filters: []*eskip.Filter{
+			{Name: spec.Name(), Args: args},
+			{Name: spec.Name(), Args: args2},
+		}, Backend: be.URL}
+		proxy := proxytest.New(fr, r)
+		defer proxy.Close()
+
+		buf := bytes.NewBufferString(content)
+		req, err := http.NewRequest("POST", proxy.URL, buf)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		rsp, err := proxy.Client().Do(req)
+		if err != nil {
+			t.Fatal(err)
+		}
+		result, _ := stdlibio.ReadAll(rsp.Body)
+		defer rsp.Body.Close()
+		if rsp.StatusCode != 200 {
+			t.Errorf("Blocked response status code %d: %s", rsp.StatusCode, string(result))
+		}
+	})
+
 	t.Run("pass request on empty body", func(t *testing.T) {
+		args := []interface{}{"foo"}
+		r := &eskip.Route{Filters: []*eskip.Filter{{Name: spec.Name(), Args: args}}, Backend: backend.URL}
+		proxy := proxytest.New(fr, r)
+		defer proxy.Close()
+
 		buf := bytes.NewBufferString("")
-		req, err := http.NewRequest("POST", reqURL.String(), buf)
+		req, err := http.NewRequest("POST", proxy.URL, buf)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		rsp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer rsp.Body.Close()
+		if rsp.StatusCode != 200 {
+			t.Errorf("Blocked response status code %d", rsp.StatusCode)
+		}
+	})
+	t.Run("pass request on empty body with filter chain", func(t *testing.T) {
+		args := []interface{}{"foo"}
+		args2 := []interface{}{"bar"}
+		r := &eskip.Route{Filters: []*eskip.Filter{
+			{Name: spec.Name(), Args: args},
+			{Name: spec.Name(), Args: args2},
+		}, Backend: backend.URL}
```

Comment:
Why not (here and elsewhere)
```go
r = eskip.MustParse(fmt.Sprintf(`block("foo") -> block("bar") -> "%s"`, backend.URL))
proxy := proxytest.New(fr, r...)
```

Reply:
It was to easily copy and paste and maybe also do a table test, but I will change as I did not do the table for different block* filters

Reply:
blockHexContent tests can not be done like this, that's why I did both test cases similar.